src/app/navbar/navbar.component.ts

```diff
@@ -47,4 +56,24 @@ export class NavbarComponent implements OnInit {
       this.isCollapsed = true;
     }
 
-}
+    getTemperature(callback): void{
+      var request = new XMLHttpRequest();
+      var data;
+      var temperature;
+      var icon;
+
+      request.open('GET', 'http://api.openweathermap.org/data/2.5/weather?units=imperial&zip=90024,us&APPID=bc6a73dfabbd4e6c9006a835d00589f2', true);
+      request.onreadystatechange = function(){
```

Comment:
This is actually an overcomplicated way to make an api request, just use this.http.get. You may need to add that library, but otherwise the format is like other API requests in our code